lib/govuk_index/presenters/details_presenter.rb

```diff
@@ -1,5 +1,10 @@
 module GovukIndex
   class DetailsPresenter
+    extend MethodBuilder
+    self.data_object = :details
```

Comment:
I think it's not obvious enough how `data_object` and `add_method` interact here. 

It feels a bit magic because you have to look at the code of the mixin to understand what this line does, and if I'm looking at what this presenter does I don't really want to care about `MethodBuilder`.

What I'd suggest instead is to use an argument of `add_method` to explicitly say where you are getting the data from. This method feels very similar to rails's `delegate`, which uses the named argument `to` for this.

If you're going with the `add_method` approach, what about something like this?

```ruby
delegate_to_hash :payload "licence_identifier"
delegate_to_hash :payload, public_timestamp: "public_updated_at"
```

or

```ruby
map_field :public_timestamp hash: :payload, key: "public_updated_at"
```